### PR TITLE
Move rsc_start_requirement to libcrmcommon

### PR DESCRIPTION
**Pull request by kgaillot**
This is part of a project to consolidate libpe_rules/libpe_status into libcrmcommon, updating code to current guidelines as we go

### CODE REVIEW

**Comment by kgaillot** [comment]
@wenningerk , can you review this low-priority PR? Thanks

File: include/crm/common/resources.h [hunk 1]
```diff
@@ -24,6 +24,7 @@ extern "C" {
 enum rsc_start_requirement {
     pcmk_requires_nothing   = 0,    //!< Resource can be recovered immediately
     pcmk_requires_quorum    = 1,    //!< Resource can be recovered if quorate
+    pcmk_requires_fencing   = 2,    //!< Resource can be recovered after fencing
```
Comment (wenningerk) [inline include/crm/common/resources.h:28]: Did neither check standards nor did I try but iirc it should be possible to combine explicit assignment with defaulting to increment by 1. But even if it is safe pros & cons ...

**Comment by wenningerk** [comment]
That is fine-grained patch-wise ;-) But of course looking good.

**Comment by kgaillot** [comment]
> > That is fine-grained patch-wise ;-) But of course looking good.
> 
> Yep :) It's for the commit messages to automatically be picked up for the changelog. My hope is that in the future, someone developing against the libraries can just search the changelog to be sure when a given symbol was added or removed. Of course with 20 years of not doing that, it may not be very useful for a while :)

I felt guilty griping about not being able to find that info for glib/libxml2 when we didn't do it ourselves ;)

File: include/crm/common/resources.h (hunk 1 above)
Comment (wenningerk) [reply include/crm/common/resources.h:28]: And if you want to deprecate certain values but still wanna stay compatible later on ...
And most dangerous is probably the one case that sounds appealing at first: leave spaces and
use explicit assignment whenever you want to start a new sequence ... when you've eaten up
the free space there is probably nothing that would even warn you and I don't see a quick and
simple safeguarding measure ...

**Pull request merged**